main.go

```diff
@@ -19,257 +19,91 @@ package main
-	versionFlag := getopt.BoolLong("version", 'v', "Displays the version of tfswitch")
-	defaultVersion := getopt.StringLong("default", 'd', defaultLatest, "Default to this version in case no other versions could be detected. Ex: tfswitch --default 1.2.4")
-
-	getopt.StringVarLong(&lib.PubKeyId, "public-key-id", 'k', "The ID of the public key to check the checksums against")
-	getopt.StringVarLong(&lib.PubKeyPrefix, "public-key-prefix", 'x', "The prefix of the public key. i.e. \"hashicorp_\"")
-	getopt.StringVarLong(&lib.PubKeyUri, "public-key-uri", 'y', "The URI to download the public key from")
-
-	helpFlag := getopt.BoolLong("help", 'h', "Displays help message")
-	_ = versionFlag
-
-	getopt.Parse()
-	args := getopt.Args()
-
-	homedir, err := homedir.Dir()
-	if err != nil {
-		logger.Fatalf("Unable to get home directory: %v", err)
-		os.Exit(1)
-	}
 
-	TFVersionFile := filepath.Join(*chDirPath, tfvFilename)    //settings for .terraform-version file in current directory (tfenv compatible)
-	RCFile := filepath.Join(*chDirPath, rcFilename)            //settings for .tfswitchrc file in current directory (backward compatible purpose)
-	TOMLConfigFile := filepath.Join(*chDirPath, tomlFilename)  //settings for .tfswitch.toml file in current directory (option to specify bin directory)
-	HomeTOMLConfigFile := filepath.Join(homedir, tomlFilename) //settings for .tfswitch.toml file in home directory (option to specify bin directory)
-	TGHACLFile := filepath.Join(*chDirPath, tgHclFilename)     //settings for terragrunt.hcl file in current directory (option to specify bin directory)
+	parameters := param_parsing.GetParameters()
 
 	switch {
-	case *versionFlag:
-		//if *versionFlag {
-		logger.Infof("Version: %s", version)
-	case *helpFlag:
-		//} else if *helpFlag {
-		usageMessage()
-	/* Checks if the .tfswitch.toml file exist in home or current directory
-	 * This block checks to see if the tfswitch toml file is provided in the current path.
-	 * If the .tfswitch.toml file exist, it has a higher precedence than the .tfswitchrc file
-	 * You can specify the custom binary path and the version you desire
-	 * If you provide a custom binary path with the -b option, this will override the bin value in the toml file
-	 * If you provide a version on the command line, this will override the version value in the toml file
-	 */
-	case lib.FileExists(TOMLConfigFile) || lib.FileExists(HomeTOMLConfigFile):
-		version := ""
-		binPath := *custBinPath
-		if lib.FileExists(TOMLConfigFile) { //read from toml from current directory
-			version, binPath = getParamsTOML(binPath, *chDirPath)
-		} else { // else read from toml from home directory
-			version, binPath = getParamsTOML(binPath, homedir)
-		}
-
-		switch {
-		/* GIVEN A TOML FILE, */
+	case parameters.VersionFlag:
+		fmt.Printf("Version: %s\n", version)
+	case parameters.HelpFlag:
+		param_parsing.UsageMessage()
```

Comment:
Doesn't it make sense to `os.Exit(0)` immediately?

Reply:
yep. probably saves some ms